samples/DataMovementSamples/DataMovementSamples/Samples.cs

```diff
@@ -98,17 +98,30 @@ private static async Task BlobUploadSample()
             UploadOptions options = new UploadOptions();
 
             SingleTransferContext context = new SingleTransferContext();
+            int transferSuccess = 0;
             context.SetAttributesCallbackAsync = async (source, destination) =>
             {
                 CloudBlob destBlob = destination as CloudBlob;
                 destBlob.Properties.ContentType = "image/png";
             };
-
             context.ShouldOverwriteCallbackAsync = TransferContext.ForceOverwrite;
+            context.FileTransferred += (object sender, TransferEventArgs args) =>
+            {
+                Interlocked.Increment(ref transferSuccess);
```

Comment:
I wanted to set a bool here instead of incrementing an int. But `Interlocked` just wouldn't let me do it through `Interlocked.Exchange`

Reply:
Should we use https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskcompletionsource-1?view=net-5.0 instead?